app/scripts/modules/core/src/pipeline/config/triggers/artifacts/gitrepo/GitRepoArtifactEditor.tsx

```diff
@@ -0,0 +1,111 @@
+import { cloneDeep, get } from 'lodash';
+import * as React from 'react';
+
+import { ArtifactTypePatterns } from 'core/artifact';
+import { IArtifactEditorProps, IArtifactKindConfig } from 'core/domain';
+import { StageConfigField } from 'core/pipeline';
+import { SpelText } from 'core/widgets';
+import { CheckboxInput } from '../../../../../presentation/forms/inputs';
+
+const TYPE = 'git/repo';
+
+interface IGitRepoArtifactEditorState {
+  includesSubPath: boolean;
+}
+
+class GitRepoArtifactEditor extends React.Component<IArtifactEditorProps, IGitRepoArtifactEditorState> {
+  public constructor(props: IArtifactEditorProps) {
+    super(props);
+    const { artifact } = this.props;
+    this.state = {
+      includesSubPath: get(artifact, 'metadata.subPath', '') !== '',
+    };
+  }
+
+  private onIncludesSubPathChange = (event: any) => {
+    this.setState({ includesSubPath: event.target.checked });
+  };
+
+  private onReferenceChanged = (reference: string) => {
+    const clonedArtifact = cloneDeep(this.props.artifact);
+    clonedArtifact.reference = reference;
+    this.props.onChange(clonedArtifact);
+  };
+
+  private onVersionChanged = (version: string) => {
+    const clonedArtifact = cloneDeep(this.props.artifact);
+    clonedArtifact.version = version;
+    this.props.onChange(clonedArtifact);
+  };
+
+  private onSubPathChanged = (subPath: string) => {
+    const clonedArtifact = cloneDeep(this.props.artifact);
+    const metadata = get(clonedArtifact, 'metadata', {}) as any;
+    metadata.subPath = subPath;
+    clonedArtifact.metadata = metadata;
+    this.props.onChange(clonedArtifact);
+  };
+
+  public render() {
+    const { artifact } = this.props;
+
+    return (
+      <>
+        <StageConfigField label="URL" helpKey="pipeline.config.expectedArtifact.gitrepo.url">
```

Comment:
Just a heads up that in new stages, we are trying to move to `FormikStageConfig` / `FormikFormField` rather than `StageConfigField` -- see `ScriptStageConfig.tsx` for an example.

Reply:
Ah, good to know! I'll check out that example, thanks. Is this something we want changed now or refactored after the merge?

Reply:
We can refactor after the merge, I think the rest of the Bake (Manifest) stage is still written with `StageConfigField` so we'd probably want to swap them all out at once in a later commit so the styling is consistent.